lib/index.js

```diff
@@ -307,28 +307,29 @@ InsightAPI.prototype.blockEventHandler = function(hashBuffer) {
   }
 };
 InsightAPI.prototype.transactionEventHandler = function(txBuffer) {
-  var tx = new Transaction().fromBuffer(txBuffer);
-  var result = this.txController.transformInvTransaction(tx);
-
-  setTimeout(function(result, txBuffer) { // delay tx relay for 900ms to see if we receive transaction lock
-    var hash = dashcore.crypto.Hash.sha256sha256(txBuffer);
-    var id = hash.toString('binary');
-
-    if (this.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
-      result.txlock = true;
-    }
+  try {
+    var tx = new Transaction().fromBuffer(txBuffer);
+    var result = this.txController.transformInvTransaction(tx);
 
     for (var i = 0; i < this.subscriptions.inv.length; i++) {
       this.subscriptions.inv[i].emit('tx', result);
     }
-  }.bind(this, result, txBuffer), 900);
+  }
+  catch(error) {
+    console.log(`${error}\nbad txBuffer: ${txBuffer.toString('hex')}`);
```

Comment:
What if one of the listeners will throw an error? Buffer might be ok in this situation. I would put only `fromBuffer` in `try` block.

Reply:
agree. Done!